spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -407,6 +409,34 @@ object TypeParser {
                                         isArrayElement: Boolean)
                                        (implicit defaults: Defaults) extends ScalarParser[String]
 
+  private final case class BinaryParser(field: TypedStructField,
+                                        path: String,
+                                        column: Column,
+                                        origType: DataType,
+                                        failOnInputNotPerSchema: Boolean,
+                                        isArrayElement: Boolean)
+                                       (implicit defaults: Defaults) extends PrimitiveParser[Array[Byte]] {
+    override protected def assemblePrimitiveCastLogic: Column = {
+      origType match {
+        case BinaryType => column
+        case StringType =>
+          val encoding = field.structField.getMetadataString(MetadataKeys.Encoding).map(_.toLowerCase)
```

Comment:
So your suggestion is to force lowercasing in validation and then rely on this?
I was trying to be more flexible and letting users use e.g. "Base64", too.

Or am I misunderstanding?

Reply:
My suggestion is to read and normalize the encoding metadata value in `BinaryTypeStructField`. Then verify in validator.
Here just use the `BinaryTypeStructField` normalized value.
The advanced option is to use enumeration in `BinaryTypeStructField`, but that can prove clumsy with validation involved.

Reply:
Changed as suggested, I hope that is what you had in mind (basically I am reusing the encoding reading and the validation as well, on the other hand, the actual application of the encoding is not reused, because it operates on different types (fromString conversion vs Colum->Column conversion)